Framework/CurveFitting/src/TableWorkspaceDomainCreator.cpp

```diff
@@ -212,13 +212,15 @@ void TableWorkspaceDomainCreator::createDomain(
     throw std::runtime_error("Workspace contains no data.");
   }
 
-  auto X = static_cast<DataObjects::TableColumn<double> &>(
-      *m_tableWorkspace->getColumn(m_xColName));
-  auto XData = X.data();
+  auto X = m_tableWorkspace->getColumn(m_xColName);
+  std::vector<double> XData;
```

Comment:
I think it is worth calling `xData.reserve(m_tableWorkspace->rowCount())` here as we know the size and we can avoid repeated memory allocation. There are a few other places in the changes that would benefit from this too. 

Also a really minor point that the coding standards mean that this should be called `xData` and not `XData`...